src/de/map.rs

```diff
@@ -62,16 +79,20 @@ impl<'de, 'a, R: BorrowingReader<'de>> MapAccess<'de, 'a, R> {
 impl<'de, 'a, R: BorrowingReader<'de> + 'a> de::MapAccess<'de> for MapAccess<'de, 'a, R> {
     type Error = DeError;
 
+    fn size_hint(&self) -> Option<usize> {
+        self.size_hint.clone()
+    }
+
```

Comment:
This is not what the `size_hint()` method is intended. It should return count of elements that deserializer _have_, not count of elements that deserialized type _requested_ from it. In our case default implementation (that returns `None`) is only possible solution because XML reader can't say how many elements it have unless it parses them all